### PR TITLE
Sentinel integration tests

### PR DESCRIPTION
**Pull request by DanielLiu1123**
Sentinel integration tests support.
See #2705

### CODE REVIEW

File: spring-cloud-alibaba-tests/sentinel-tests/sentinel-flowcontrol-test/src/main/java/com/alibaba/cloud/tests/sentinel/degrade/SentinelFlowControlTestApp.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.event.EventListener;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * @author Freeman
+ * @date 2022/8/27
```
Comment (steverao) [inline spring-cloud-alibaba-tests/sentinel-tests/sentinel-flowcontrol-test/src/main/java/com/alibaba/cloud/tests/sentinel/degrade/SentinelFlowControlTestApp.java]: Don't include the date information.

File: spring-cloud-alibaba-tests/sentinel-tests/sentinel-flowcontrol-test/pom.xml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <parent>
+        <artifactId>sentinel-tests</artifactId>
```
Comment (steverao) [inline spring-cloud-alibaba-tests/sentinel-tests/sentinel-flowcontrol-test/pom.xml]: Please adjust the order of artifactId and groupId.

File: spring-cloud-alibaba-tests/sentinel-tests/sentinel-degrade-test/src/test/java/com/alibaba/cloud/tests/sentinel/degrade/SentinelDegradeTestAppTest.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2012-2020 the original author or authors.
```
Comment (steverao) [inline spring-cloud-alibaba-tests/sentinel-tests/sentinel-degrade-test/src/test/java/com/alibaba/cloud/tests/sentinel/degrade/SentinelDegradeTestAppTest.java]: Please keep consistent with other license information, such as the copyright is 2012-2020 here, and another file is 2013-2022.

File: spring-cloud-alibaba-tests/sentinel-tests/sentinel-flowcontrol-test/src/test/java/com/alibaba/cloud/tests/sentinel/degrade/SentinelFlowControlTestAppTest.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+@SpringBootTest(webEnvironment = RANDOM_PORT)
+class SentinelFlowControlTestAppTest {
+
+	@LocalServerPort
+	int port;
+
+	@Test
+	public void testFlowControl() throws InterruptedException {
+		RestTemplate rest = new RestTemplate();
```
Comment (steverao) [inline spring-cloud-alibaba-tests/sentinel-tests/sentinel-flowcontrol-test/src/test/java/com/alibaba/cloud/tests/sentinel/degrade/SentinelFlowControlTestAppTest.java]: Maybe I think it's better that set threadCount is 2, and check if there is fallback signal in result set first. Secondly, call a method again, and check is there is fallback signal in result set. Only in this way, we can conclude the flow control threshold of 2 is valid.

**Comment by DanielLiu1123** [comment]
Still have problem to be fixed, need digging ......

**Review by kaori-seasons — APPROVED** [review]

**Review by steverao — APPROVED** [review]

**Comment by steverao** [comment]
@DanielLiu1123 Thanks!

**Pull request merged**